### PR TITLE
scheduler: add aduit log for scheduler config API and add resp msg for evict-leader

### LINKED ISSUE #7672: The output is null when adding evict-leader-scheduler for second store
### What did you do?
add evict leader scheduler for two stores.

### What did you expect to see?
Both output scheduler creation

### What did you see instead?
image

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: Close #7672

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
image
> [2024/01/08 11:20:31.719 +08:00] [INFO] [audit.go:126] ["audit log"] [service-info="{ServiceLabel:HandleSchedulerConfig, Method:HTTP/1.1/DELETE:/pd/api/v1/scheduler-config/evict-leader-scheduler/delete/2, CallerID:pd-ctl, IP:127.0.0.1    , Port:58671, StartTime:2024-01-08 11:20:31 +0800 CST, URLParam:{}, BodyParam:}"]
> [2024/01/08 11:21:06.896 +08:00] [INFO] [audit.go:126] ["audit log"] [service-info="{ServiceLabel:HandleSchedulerConfig, Method:HTTP/1.1/POST:/pd/api/v1/scheduler-config/evict-leader-scheduler/config, CallerID:pd-ctl, IP:127.0.0.1, Po    rt:58851, StartTime:2024-01-08 11:21:06 +0800 CST, URLParam:{}, BodyParam:{\"name\":\"evict-leader-scheduler\",\"store_id\":2}}"]
Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/api/router.go [hunk 1]
```diff
@@ -151,7 +151,8 @@ func createRouter(prefix string, svr *server.Server) *mux.Router {
 	registerFunc(clusterRouter, "/schedulers/diagnostic/{name}", diagnosticHandler.GetDiagnosticResult, setMethods(http.MethodGet), setAuditBackend(prometheus))
 
 	schedulerConfigHandler := newSchedulerConfigHandler(svr, rd)
-	registerPrefix(apiRouter, "/scheduler-config", schedulerConfigHandler.GetSchedulerConfig, setAuditBackend(prometheus))
+	registerPrefix(apiRouter, "/scheduler-config", "HandleSchedulerConfig", schedulerConfigHandler.HandleSchedulerConfig, setMethods(http.MethodPost, http.MethodDelete, http.MethodPut, http.MethodPatch), setAuditBackend(localLog, prometheus))
```
Comment (rleungx) [inline server/api/router.go:154]: How about SetSchedulerConfig?

Comment (CabinfeverB) [reply server/api/router.go:154]: It also includes the deletion method.

**Review by rleungx — APPROVED** [review]

File: server/api/router.go (hunk 1 above)
Comment (rleungx) [reply server/api/router.go:154]: got

**Review by nolouch — APPROVED** [review]

**Pull request merged**